File: C++/ReverseLinkedList.cpp
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* dummy = new ListNode(0);
        
        while(head != NULL) {
            ListNode* next = head->next;
            
            ListNode* tmp = dummy->next;
            dummy->next = head;
            head->next = tmp;
            
            head = next;
        }
        
        return dummy->next;
    }
};

File: Java/ReverseLinkedList.java
public class Solution {
    public ListNode reverseList(ListNode head) {
        ListNode rList = null;
        
        while(head != null) {
            ListNode tmp = head.next;
            
            head.next = rList;
            rList = head;
            head = tmp;
        }
        
        return rList;
    }
}